boards/arm/ti_am62x_sk_m4/ti_am62x_sk_m4_defconfig

```diff
@@ -0,0 +1,20 @@
+# Texas Instruments Sitara AM62x-SK-M4 EVM
+#
+# Copyright (c) 2023 L Lakshmanan
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Platform Configuration
+CONFIG_SOC_SERIES_AM62X_M4=y
+CONFIG_SOC_AM62x_M4=y
+CONFIG_BOARD_TI_AM62X_SK_M4=y
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_SEMIHOST=y
+
+# Zephyr Kernel Configuration
+CONFIG_XIP=n
```

Comment:
any particular reason to set `CONFIG_XIP=n` ? Can you try and remove this line and check if it works?

Reply:
> I can see that you have added `soc/arm/ti_k3` [ca7c3a2](https://github.com/zephyrproject-rtos/zephyr/commit/ca7c3a23fb9d1f269ecfd43e701236195f2e72f8) and then in [a72e4e0](https://github.com/zephyrproject-rtos/zephyr/commit/a72e4e0a2fc2183d94bec62a2905a95f6e5ed530) you are editing some of the files you yourself had added in the earlier commit. This is in general a practise to **avoid**, especially since you are adding completely new files in your PR, you can just add them with the necessary changes at once, not in 2 seperate commits unless there's a very good reason for doing so? I think that is one reason also why compliance checks maybe failing, although I maybe wrong about that.

Yes, I observed that too, have rebased and fixed it now.

Reply:
> any particular reason to set `CONFIG_XIP=n` ? Can you try and remove this line and check if it works?

Right now, the build fails as it overflows, due to the ram being very small. Will attempt after adding access to more memory.